### ISSUE TITLE
Proposal Generator using a Pre-trained Region Proposal Network?

### ISSUE THREAD

**Issue by aligholami** [facebookresearch/detectron2#628]
I wonder how I can use a pre-trained region proposal network to generate a bunch of proposal boxes for a given image feature map? I'm going through the test code in [rpn test](https://github.com/facebookresearch/detectron2/blob/master/tests/test_rpn.py), and I would like to generate a bunch of boxes, without having the ground truth for my dataset.
Thanks!

**Comment by ppwwyyxx** [comment]
See https://detectron2.readthedocs.io/tutorials/models.html#use-models and use a RPN config, such as https://github.com/facebookresearch/detectron2/blob/master/configs/COCO-Detection/rpn_R_50_FPN_1x.yaml

**Comment by aligholami** [comment]
@ppwwyyxx Thanks! I'm already using an RPN config, but I guess the following lines do something different than that:
```
        with EventStorage():  # capture events in a new storage to discard them
            proposals, proposal_losses = proposal_generator(
                images, features, [gt_instances[0], gt_instances[1]]
            )
```

Where do I get gt_instances if don't have them at all?

**Comment by ppwwyyxx** [comment]
I'm not sure where you get those lines. The docs say:

> Then call outputs = model(inputs). For example, in order to do inference, provide dicts with “image”, and optionally “height” and “width”.

maybe you forgot to set your model in eval mode with `model.eval()`.

**Comment by aligholami** [comment]
@ppwwyyxx Here: https://github.com/facebookresearch/detectron2/blob/02431913b5fb01bc20b15ae683147f9bbabb8583/tests/test_rpn.py#L34

**Comment by ppwwyyxx** [comment]
The file is testing training ability and therefore is not relevant to what you're trying to do.

**Comment by aligholami** [comment]
I think I put my question in the wrong way.
I am writing my own class to generate a region feature matrix for a bunch of images. This class is like:
```
def ObjectDetector(nn.Module):
    def __init__(self, cfg):
        ...
    def forward(self, x):
        ...
```
and I would like to perform API calls to the following functions:
1. `build_backbone`: To build a CNN backbone using the information in `cfg`.
2. `build_proposal_generator`: To build a proposal generator using the information in `cfg`.
3. `ROIPooler`: To perform pooling operation on the generated boxes in step 2.

And finally returning the region feature matrix as the output of my forward function. Do you find this plausible?

Thanks for your kind replies!

**Comment by aligholami** [comment]
This approach is motivated by another issue (#5) and the information provided by you in this comment (https://github.com/facebookresearch/detectron2/issues/5#issuecomment-540932497). However, I'm not sure what would be the inputs to the ROIPooler, that's why I'm using these APIs to generate the proposals to pass them to my ROIPooler.

**Comment by ppwwyyxx** [comment]
```
proposals, _ = proposal_generator(images, features)
boxes = [x.proposal_boxes for x in proposals]
# boxes can be inputs to ROIPooler
```

**Comment by aligholami** [comment]
That's it! Sorry that I didn't read the docs for `proposal_generator` thoroughly.
